web/src/components/task/task-sidebar/task-sidebar.tsx

```diff
@@ -223,6 +234,35 @@ const TaskSidebar: FC<TaskSidebarProps> = ({ onRedirectAfterFinish, jobSettings,
         isValid ? styles.validColor : styles.invalidColor
     }`;
     const validationStatus: ValidationPageStatus = isValid ? 'Valid Page' : 'Invalid Page';
+    useEffect(() => {
+        refetchTaxons();
+    }, [latestLabelsId]);
+
+    const taxonomyFilter: Filter<keyof Taxon> = {
+        field: 'id',
+        operator: Operators.IN,
+        value: latestLabelsId
+    };
+
+    const { data: latestTaxons, refetch: refetchTaxons } = useTaxonomies(
```

Comment:
Taxonomies and Taxons are different entities, useTaxonomies should be changed to useTaxons